impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -200,6 +207,91 @@ Rect MakeReferenceUVs(const Rect& reference, const Rect& rect) {
+};
+
+std::shared_ptr<Geometry> GetGeometry(
+    const std::shared_ptr<FilterInput>& input) {
+  std::shared_ptr<Geometry> result;
+  std::visit(
+      [&result](auto&& arg) {
+        using T = std::decay_t<decltype(arg)>;
+        if constexpr (std::is_same_v<T, std::shared_ptr<FilterContents>>) {
+          GeometryExtractor extractor;
+          arg->Visit(&extractor);
+          result = extractor.geometry_;
+        } else if constexpr (std::is_same_v<T, std::shared_ptr<Contents>>) {
+          GeometryExtractor extractor;
+          arg->Visit(&extractor);
+          result = extractor.geometry_;
+        } else if constexpr (std::is_same_v<T, std::shared_ptr<Texture>>) {
+          // TODO (https://github.com/flutter/flutter/issues/144266): I think
+          // gaussian blurs have to work first for this to happen.
+        } else if constexpr (std::is_same_v<T, Rect>) {
+          // Does this ever happen?
+          FML_DCHECK(false) << "unimplemented";
+        } else {
+          static_assert(kAlwaysFalseV<T>, "non-exhaustive visitor!");
+        }
+      },
+      input->GetInput());
+  return result;
+}
+
+Entity ApplyBlurStyle(FilterContents::BlurStyle blur_style,
+                      const Entity& entity,
+                      const std::shared_ptr<FilterInput>& input,
+                      const Snapshot& input_snapshot,
+                      Entity blur_entity) {
+  if (blur_style == FilterContents::BlurStyle::kNormal) {
+    return blur_entity;
+  }
+  Entity::ClipOperation clip_operation;
+  switch (blur_style) {
+    case FilterContents::BlurStyle::kNormal:
+      FML_UNREACHABLE();
+      break;
+    case FilterContents::BlurStyle::kInner:
+      clip_operation = Entity::ClipOperation::kIntersect;
+      break;
+    case FilterContents::BlurStyle::kOuter:
+      clip_operation = Entity::ClipOperation::kDifference;
+      break;
+    case FilterContents::BlurStyle::kSolid:
+      FML_DLOG(ERROR) << "Unimplemented blur style";
+      return blur_entity;
+  }
+
+  std::shared_ptr<Geometry> geometry = GetGeometry(input);
+  auto shared_blur_entity = std::make_shared<Entity>(std::move(blur_entity));
+  auto clipper = std::make_unique<ClipContents>();
+  clipper->SetClipOperation(clip_operation);
+  clipper->SetGeometry(geometry);
+  auto restore = std::make_unique<ClipRestoreContents>();
+  Entity result;
+  result.SetTransform(entity.GetTransform());
+  result.SetContents(Contents::MakeAnonymous(
+      fml::MakeCopyable([shared_blur_entity, clipper = std::move(clipper),
+                         restore = std::move(restore)](
+                            const ContentContext& renderer,
+                            const Entity& entity, RenderPass& pass) mutable {
+        bool result = true;
+        result = clipper->Render(renderer, entity, pass) && result;
+        result = shared_blur_entity->Render(renderer, pass) && result;
+        result = restore->Render(renderer, entity, pass) && result;
```

Comment:
This restore can be made a lot faster in some cases by restricting the restore to the clip's geometry coverage:

```cpp
restore->SetRestoreCoverage(geometry->GetCoverage(entity->GetTransform());
```

Reply:
It sounds like we won't need this though shortly?  I'd like to do that optimization as a separate PR since I've seen a lot of bugs related to Coverage, so I'm nervous predicating rendering on it without test coverage.  That's moot if we plan on removing it anyways.  Let me know if you want me to file a followup issue.

Reply:
Yeah deferring this SGTM. 👍